--- a/greeter/pamauthenticator.cpp
+++ b/greeter/pamauthenticator.cpp
@@ -91,7 +91,9 @@
     auto messages = std::span{msg, nSize};
     Q_ASSERT_X(responses.size() == messages.size(), Q_FUNC_INFO, "Number of PAM messages and responses should be the same");
 
-    for (const auto &[pamMessage, pamResponse] : std::views::zip(messages, responses)) {
+    for (size_t index = 0; index < messages.size(); ++index) {
+        const auto &pamMessage = messages[index];
+        auto &pamResponse = responses[index];
         bool isSecret = false;
         switch (pamMessage->msg_style) {
         case PAM_PROMPT_ECHO_OFF: {
